### ISSUE TITLE
[gemma4][fsdp] Gemma-4 Training Issues: Exploding Gradients (4B) and FQN Mismatch Loading Failure (31B) with FSDP

### ISSUE THREAD

**Issue by mamazi0131** [verl-project/verl#5999]
### System Info

We observed two distinct and critical issues when adapting the Gemma-4 model family (E4B-it and 31B-it) within the verl framework using FSDP under on-policy setting. 

Gemma-4-E4B: The model initializes and runs, but the gradient norm is abnormally high (consistently > 10.0, peaking at 50+), leading to training instability.

[swanlab](https://swanlab.cn/@allenzpma/verl_grpo_gemma_4b/runs/6p23z3da3vxf9l3k47ymn/chart)

Gemma-4-31B: The model fails to load entirely. During the load_state_dict phase in SHARDED_STATE_DICT mode, FSDP skips all parameters due to a mismatch between the expected nested FQNs (Fully Qualified Names) and the keys in the checkpoint.

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] An officially supported task in the `examples` folder (such as GLUE/SQuAD, ...)
- [x] My own task or dataset (give details below)

### Reproduction

My startup script is as follows：

```bash
export VLLM_USE_V1=1
export VLLM_ALLREDUCE_USE_SYMM_MEM=0
export SWANLAB_API_KEY=xxxxxx
export SWANLAB_MODE=cloud

echo $NODE_IP_LIST > env.txt
sed "s/:/ slots=/g" env.txt | sed "s/,/\n/g" > "hostfile"
sed "s/:.//g" env.txt | sed "s/,/\n/g" > "pssh.hosts"

source ~/.bashrc

bash stop_ray.sh
bash start_ray.sh

########################### Quick Config ###########################

GEN_TP=${GEN_TP:-2}
ALL_OFFLOAD=${ALL_OFFLOAD:-True}

rollout_name="vllm"
project_name='verl_grpo_gemma_4b'
exp_name='gemma_4b_fsdp_text_grpo'
OUTPUT_DIR=/apdcephfs_bjzf/share_304704649/allenzpma/deliver/output
CKPTS_DIR=${OUTPUT_DIR}/"${project_name}"/"${exp_name}/ckpts"
ROLLOUT_DATA_DIR=${OUTPUT_DIR}/"${project_name}"/"${exp_name}/rollout_data"
VALIDATION_DATA_DIR=${OUTPUT_DIR}/"${project_name}"/"${exp_name}/validation_data"
LOG_DIR="${OUTPUT_DIR}/${project_name}/${exp_name}/logs/$(date +%Y%m%d_%H%M%S)"
SWANLAB_LOG_DIR=${OUTPUT_DIR}/"${project_name}"/"${exp_name}/swanlog"

mkdir -p ${CKPTS_DIR}
mkdir -p ${ROLLOUT_DATA_DIR}
mkdir -p ${VALIDATION_DATA_DIR}
mkdir -p ${LOG_DIR}
mkdir -p ${SWANLAB_LOG_DIR} 

adv_estimator=grpo
NNODES=2
max_prompt_length=1024
max_response_length=2048
train_prompt_bsz=64
train_prompt_mini_bsz=${train_prompt_bsz}

train_temperature=0.8
train_rollot_n=8

HF_MODEL_PATH=${HF_MODEL_PATH:-/xxx/model/google/gemma-4-E4B-it/hf} 
DATASET_DIR=${DATASET_DIR:-/xxx/data}
train_path=${train_path:-${DATASET_DIR}/rl/gsm8k/train.parquet}
test_path=${test_path:-${DATASET_DIR}/rl/gsm8k/test.parquet}

########################### Parameter Arrays ###########################

DATA=(
    data.train_files=${train_path}
    data.val_files=${test_path}
    data.train_batch_size=${train_prompt_bsz}
    data.max_prompt_length=${max_prompt_length}
    data.max_response_length=${max_response_length}
    data.truncation='error'
    data.filter_overlong_prompts=True
)

MODEL=(
    actor_rollout_ref.model.path=${HF_MODEL_PATH}
    actor_rollout_ref.model.trust_remote_code=True
    actor_rollout_ref.model.use_remove_padding=True
    +actor_rollout_ref.model.override_config.attn_implementation=sdpa
)

ACTOR=(
    actor_rollout_ref.actor.strategy=fsdp
    actor_rollout_ref.actor.optim.lr=1e-6
    actor_rollout_ref.actor.ppo_mini_batch_size=${train_prompt_mini_bsz}
    actor_rollout_ref.actor.ppo_micro_batch_size_per_gpu=4
    actor_rollout_ref.actor.use_kl_loss=False
    actor_rollout_ref.actor.kl_loss_coef=0.01
    actor_rollout_ref.actor.fsdp_config.param_offload=${ALL_OFFLOAD}
    actor_rollout_ref.actor.fsdp_config.optimizer_offload=${ALL_OFFLOAD}
    +actor_rollout_ref.actor.fsdp_config.grad_offload=${ALL_OFFLOAD}
    actor_rollout_ref.actor.grad_clip=1.0
)

ROLLOUT=(
    actor_rollout_ref.rollout.name=${rollout_name}
    actor_rollout_ref.rollout.tensor_model_parallel_size=${GEN_TP}
    actor_rollout_ref.rollout.gpu_memory_utilization=0.5
    actor_rollout_ref.rollout.n=${train_rollot_n}
    actor_rollout_ref.rollout.temperature=${train_temperature}
    actor_rollout_ref.rollout.dtype=bfloat16
    actor_rollout_ref.rollout.log_prob_micro_batch_size_per_gpu=2
    actor_rollout_ref.rollout.checkpoint_engine.update_weights_bucket_megabytes=6000
)

REF=(
    actor_rollout_ref.ref.strategy=fsdp
    actor_rollout_ref.ref.fsdp_config.param_offload=${ALL_OFFLOAD}
    actor_rollout_ref.ref.log_prob_micro_batch_size_per_gpu=2
)

ALGORITHM=(
    algorithm.adv_estimator=${adv_estimator}
    algorithm.use_kl_in_reward=False
)

TRAINER=(
    trainer.critic_warmup=0
    trainer.logger='["console","swanlab"]'
    trainer.project_name=${project_name}
    trainer.experiment_name=${exp_name}
    trainer.default_local_dir=${CKPTS_DIR}
    trainer.n_gpus_per_node=8
    trainer.nnodes=${NNODES}
    trainer.save_freq=50
    trainer.test_freq=5
    trainer.total_epochs=15
)

RAY_KWARGS=(
    +ray_kwargs.ray_init.runtime_env.env_vars.SWANLAB_LOG_DIR="${SWANLAB_LOG_DIR}"
    +ray_kwargs.ray_init.runtime_env.env_vars.SWANLAB_MODE="${SWANLAB_MODE}"
    +ray_kwargs.ray_init.runtime_env.env_vars.SWANLAB_API_KEY="${SWANLAB_API_KEY}"
)

########################### Launch ###########################

python3 -m verl.trainer.main_ppo \
    --config-path=config \
    --config-name='ppo_trainer.yaml' \
    "${DATA[@]}" \
    "${ALGORITHM[@]}" \
    "${MODEL[@]}" \
    "${ROLLOUT[@]}" \
    "${ACTOR[@]}" \
    "${REF[@]}" \
    "${TRAINER[@]}" \
    "${RAY_KWARGS[@]}" \
    "$@" 2>&1 | tee "$LOG_DIR/train.log"

bash stop_ray.sh
```

### Expected behavior

Fix the abnormally high gradient issue to stabilize training and support gemma4 31b.

**Comment by mamazi0131** [comment]
The relevant issues have been resolved
